apps/aecontract/src/aect_dispatch.erl

```diff
@@ -160,3 +161,17 @@ error_to_binary(out_of_stack) -> <<"out_of_stack">>;
 error_to_binary(E) ->
     ?DEBUG_LOG("Unknown error: ~p\n", [E]),
     <<"unknown_error">>.
+
+chain_state(#{ contract    := ContractPubKey
+             , trees       := Trees} = CallDef, TxEnv) ->
+    case maps:get(off_chain, CallDef, false) of
```

Comment:
Avoiding default would make the code more readable, by making explicit the point of the decision about off/on chain.